### PR TITLE
[Continuous batching] Replace standard max_element call with custom loop for greedy sampling

### PR DESCRIPTION
**Pull request by mzegla**
Searching for max element in a custom loop gives better performance than using std::max_element

### CODE REVIEW

File: src/cpp/src/sampler.hpp [hunk 1]
```diff
@@ -219,8 +219,13 @@ class Sampler {
     }
 
     Token _greedy_sample(const std::vector<Token>& logit_vector) const {
-        auto out_token = std::max_element(logit_vector.begin(), logit_vector.end(), [](const Token& lhs, const Token& rhs) { return lhs.m_log_prob < rhs.m_log_prob; });
-        return *out_token;
+        Token max_token{0.0, 0};
```
Comment (Wovchena) [inline src/cpp/src/sampler.hpp:222]: I believe logits can be negative. -inf should be a good initial value.

**Review by Wovchena — APPROVED** [review]
